### PR TITLE
TD-3232 Create variations of `IconWithLabel`

### PR DESCRIPTION
**Pull request by AronPetrauskas**
Closes [TD-3232](https://sce.myjetbrains.com/youtrack/issue/TD-3232)
Contributes [TD-3196](https://sce.myjetbrains.com/youtrack/issue/TD-3196)

## Changes

>[!NOTE]
>Even though the code diff looks big, actually a lot of the components and their test are very repetitive and should be simple to review. All components except `TruckMakerLogo` and `RoadOutlined` are just wrappers of the `IconWithLabel` component. These wrappers are very simplistic and they just provide a different logo to `IconWithLabel`. 

- Added the following components: `RoadLabel`, `FormatLabel`, `FormatVersionLabel`, `VersionLabel`, `FileLabel`. `DateLabel`, `UserLabel`, `RoadOutlined` and `TruckMakerLogo`
- Added tests for all of the above and `CarMakerLogo`

## UI/UX

`RoadLabel`:
image

`FormatLabel`:
image

`FormatVersionLabel`:
image

`VersionLabel`:
image

`FileLabel`:
image

`DateLabel`:
image

`UserLabel`:
image

`RoadOutlined`:
image

`TruckMakerLogo`:
image

## Testing notes

>[!NOTE]
> Truncation is tested on the main `IconWithLabel` component, thus we know that the label is able to truncate and hence there are no truncation tests on the wrappers.

[FIGMA](https://www.figma.com/design/WoHtIQKM6eLlOUuzK1zlpB/VIRTO.TEST?node-id=3593-28235&node-type=instance&t=Cw9lPqpysUMeGhnO-0)

For testing load up storybook and check the following.

`RoadLabel`:
- Shows the road name you pass
- Accepts a link, on click it opens this link in a new tab
- Shows the `RoadOutlined` icon to the left
- Matches FIGMA

 `FormatLabel`:
- For the strings `CarMaker`, `CM4SL`, `TruckMaker`, `TM4SL`, `ASAM OpenDRIVE`, `ASAM OpenSCENARIO XML` a relevant icon is displayed.
- For custom formats a default icon is displayed to the left
- Matches FIGMA

 `FormatVersionLabel`:
- Accepts any string and displays it (it is on purpose it is not limited to numbers)
- Matches FIGMA

 `VersionLabel`:
- Accepts any string and displays it (it is on purpose it is not limited to numbers)
- Matches FIGMA

 `FileLabel`:
- Displays the name/label passed
- Matches FIGMA

 `DateLabel`:
- Accepts any string and displays it (it is on purpose it is not limited to numbers)
- Matches FIGMA

`UserLabel`:
- Accepts any string and displays it to the right of the Avatar
- The Avatar displays the initials of the label passed (e.g., James Harper --> JH)

`RoadOutlined`:
- RoadIcon renders
- Accepts sx to adjust style
- Matches FIGMA

`TruckMakerLogo`:
- TruckMakerLogo renders
- Accepts sx to adjust style
- Matches FIGMA

## Author checklist

Before I request a review:

- [x] I have reviewed my own code-diff.
- [x] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [x] I have included appropriate tests.
- [x] I have checked that the Lint and Test workflows pass on Github.
~- [ ] I have populated the deploy-preview with relevant test data.~
- [x] I have tagged a UI/UX designer for review if this PR includes UI/UX changes.

### CODE REVIEW

**Comment by AronPetrauskas** [comment]
@Sowbhagya-ipg Can you please approve that the components in this PR match your expectations?

File: src/UserLabel/UserLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Component that renders a user avatar and the name of the user to the right
+ * @param label The name of the user to be displayed
+ */
+export default function UserLabel({ label }: UserLabelProps) {
+  return IconWithLabel({
+    icon: (
+      <div>
+        <UserAvatar
+          sx={{ height: "24px", width: "24px" }}
+          color="#EC407A"
```
Comment (AronPetrauskas) [inline src/UserLabel/UserLabel.tsx:16]: @dmbarry86 and @Sowbhagya-ipg I followed the current implementation of the road preview, where colour is hard coded, is this fine for now?

Comment (dmbarry86) [reply src/UserLabel/UserLabel.tsx:16]: We should avoid hard-coding any colours inside components. Instead, we should reference them centrally using the theme. You can extend the theme if it does not currently contain a prop for what you need.

Comment (AronPetrauskas) [reply src/UserLabel/UserLabel.tsx:16]: I have exposed the prop now that the color can be controlled. If no color prop is passed, the `UserAvatar` has a default value. I have updated storybook and added tests.

**Comment by Sowbhagya-ipg** [comment]
Screenshot 2024-11-28 at 1 21 58 pm
Road name label icon - in dark mode should be in white color. @AronPetrauskas

**Review by jzhelyazkovtvm — CHANGES_REQUESTED** [review]
UI and functionality are working well. Just some suggestions left.

**Comment by AronPetrauskas** [comment]
@Sowbhagya-ipg I have fixed the `RoadOutlineIcon` not changing colour. Can you please confirm now it is okay?

**Review by jzhelyazkovtvm — APPROVED** [review]
LGTM

**Comment by Sowbhagya-ipg** [comment]
Looks good @AronPetrauskas

**Pull request merged**